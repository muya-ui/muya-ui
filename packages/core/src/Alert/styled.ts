import styled, { css } from 'styled-components';

import { IComponentSizeSpec } from '@muya-ui/theme-light';

import { StyledButton, StyledLink } from '../Button/styled';
import { IThemedBaseProps } from '../types';
import { IAlertProps } from './types';

interface IBaseSizeProps {
  $size: IComponentSizeSpec;
}

export interface IStyledAlertProps extends IThemedBaseProps, IBaseSizeProps {
  $type: Required<IAlertProps>['type'];
  $hasTitle?: boolean;
  $hasDesc?: boolean;
  $showIcon?: boolean;
  $showClose?: boolean;
}

export interface IStyledIconProps extends IThemedBaseProps, IBaseSizeProps {
  $type: Required<IAlertProps>['type'];
  $hasTitleAndDesc?: boolean;
}

export interface IStyledCloseProps extends IThemedBaseProps, IBaseSizeProps {
  $hasTitleAndDesc?: boolean;
}

export interface IStyledTitleProps extends IThemedBaseProps, IBaseSizeProps {
  $hasDesc?: boolean;
}

export const StyledAlert = styled.div`
  ${(props: IStyledAlertProps) => {
    const { $type, $hasTitle, $hasDesc, $showIcon, $showClose, $size, theme } = props;
    const {
      typeBg,
      typeBorder,
      borderRadius,
      padding,
      hasIconPaddingLeft,
      hasIconAndDescPaddingLeft,
      hasCloseIconPaddingRight,
      hasTitleAndDescPaddingVertical,
    } = theme.components.Alert;
    let [top, right, bottom, left] = padding[$size];
    if ($showIcon) {
      left = $hasDesc ? hasIconAndDescPaddingLeft[$size] : hasIconPaddingLeft[$size];
    }
    if ($showClose) {
      right = hasCloseIconPaddingRight[$size];
    }
    if ($hasTitle && $hasDesc) {
      top = hasTitleAndDescPaddingVertical[$size];
      bottom = hasTitleAndDescPaddingVertical[$size];
    }
    return css`
      position: relative;
      box-sizing: border-box;
      background-color: ${typeBg[$type!]};
      border: ${typeBorder[$type!]};
      border-radius: ${borderRadius};

      /** 直接改 Button 存在风险，因此覆盖 */
      & ${StyledButton}, & ${StyledLink} {
        vertical-align: unset;
      }

      ${StyledAlertContainer} {
        padding: ${top}px ${right}px ${bottom}px ${left}px;
      }
    `;
  }}
`;

export const StyledAlertContainer = styled.div`
  position: relative;
  margin: 0 auto;
`;

export const StyledIcon = styled.div`
  ${(props: IStyledIconProps) => {
    const { $type, theme, $hasTitleAndDesc, $size } = props;
    const {
      left: iconLeft,
      hasTitleAndDescLeft,
      top: iconTop,
      hasTitleAndDescTop,
    } = theme.components.Alert.icon;
    const { lineHeight } = theme.components.Alert.title;
    const finalLineHeight = $hasTitleAndDesc ? 1 : `${lineHeight[$size]}px`;
    const left = $hasTitleAndDesc ? hasTitleAndDescLeft : iconLeft[$size];
    const top = $hasTitleAndDesc ? hasTitleAndDescTop : iconTop[$size];
    return css`
      position: absolute;
      left: ${left}px;
      top: ${top}px;
      color: ${theme.colors.pattern.feature[$type!]};
      font-size: 0;
      line-height: ${finalLineHeight};
    `;
  }}
`;

export const StyledClose = styled.div`
  ${(props: IStyledCloseProps) => {
    const { theme, $hasTitleAndDesc, $size } = props;
    const { closeIconTop, closeIconRight, hasTitleAndDescCloseIconTop } = theme.components.Alert;
    let finalTop = closeIconTop[$size];
    if ($hasTitleAndDesc) {
      finalTop = hasTitleAndDescCloseIconTop;
    }
    return css`
      position: absolute;
      right: ${closeIconRight}px;
      top: ${finalTop}px;
      line-height: 0;
    `;
  }}
`;

export const StyledTitle = styled.div`
  ${(props: IStyledTitleProps) => {
    const { theme, $size, $hasDesc } = props;
    const {
      colors,
      typography: {
        spec: { fontWeight },
      },
    } = theme;
    const {
      fontSize,
      lineHeight,
      textFine,
      hasDescTextFine,
      hasDescFontSize,
      hasDescLineHeight,
      hasDescMarginBottom,
    } = theme.components.Alert.title;
    const finalTextFine = $hasDesc ? hasDescTextFine : textFine;
    const finalFontSize = $hasDesc ? hasDescFontSize : fontSize[$size];
    const finalLineHeight = $hasDesc ? hasDescLineHeight : lineHeight[$size];
    const finalMarginBottom = $hasDesc ? hasDescMarginBottom : 0;
    return css`
      color: ${colors.pattern.text.title};
      font-size: ${finalFontSize}px;
      line-height: ${finalLineHeight}px;
      font-weight: ${finalTextFine ? fontWeight.regular : fontWeight.semibold};
      margin-bottom: ${finalMarginBottom}px;
    `;
  }}
`;

export const StyledDesc = styled.span`
  ${(props: IThemedBaseProps) => {
    const { theme } = props;
    const { colors } = theme;
    const { fontSize, lineHeight } = theme.components.Alert.desc;
    return css`
      color: ${colors.pattern.text.text};
      font-size: ${fontSize}px;
      line-height: ${lineHeight}px;
    `;
  }}
`;
