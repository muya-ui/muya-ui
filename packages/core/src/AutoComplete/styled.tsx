import styled, { css } from 'styled-components';

import ellipsisStyle from '../styled/mixins/ellipsis';

export const StyledAutoCompleteInputWrapper = styled.span`
  position: relative;
  display: inline-block;
  width: 100%;
`;

export const StyledPirmaryField = styled.span`
  ${ellipsisStyle};
`;

export const StyledSecondaryField = styled.span`
  flex-shrink: 0;
  margin-left: 12px;
`;

export const StyledAutoCompleteItem = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
  ${props => css`
    ${StyledSecondaryField} {
      color: ${props.theme.colors.pattern.text.assistant};
    }
  `};
`;
