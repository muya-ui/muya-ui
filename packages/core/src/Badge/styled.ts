import styled, { css } from 'styled-components';

import { IComponentSizeSpec } from '@muya-ui/theme-light';

import { IThemedBaseProps } from '../types';

export interface IStyledBadgeProps extends IThemedBaseProps {
  $detached: boolean;
  $isStroke: boolean;
  $value?: number | string;
  $showZero: boolean;
  $dot: boolean;
  $size: IComponentSizeSpec;
  $color?: string;
}

const badgePositionCSS = (props: IStyledBadgeProps) => {
  const { theme, $detached, $value, $showZero, $dot } = props;
  const {
    Badge: { offset, zIndex },
  } = theme.components;
  const visible = $value || $showZero || $dot;
  const scaleCSS = `scale(${visible ? 1 : 0})`;
  let positionCSS;
  if ($detached) {
    positionCSS = `
      top: 50%;
      left: 100%;
      transform: translate(${offset}px, -50%) ${scaleCSS};
    `;
  } else {
    positionCSS = `
      top: 0;
      right: 0;
      transform: translate(50%, -50%) ${scaleCSS};
    `;
  }

  return css`
    position: absolute;
    ${positionCSS}
    z-index: ${zIndex};
  `;
};

const badgeCoreCSS = (props: IStyledBadgeProps) => {
  const { theme, $size, $dot, $isStroke, $color } = props;
  const {
    components: { Badge: token },
    transition: {
      pattern: { duration, easing },
    },
  } = theme;

  let height;
  let padding;
  if ($dot) {
    height = token.height.dot[$size!];
    padding = 0;
  } else {
    height = token.height.text[$size!];
    padding = token.padding[$size!];
  }

  const fontSize = token.fontSize[$size!];
  const { borderColor } = token;
  const lineHeight = $isStroke ? height - 2 : height;
  const color = $isStroke ? token.borderColor : token.color;
  const backgroundColor = $color || token.backgroundColor[$isStroke ? 'stroke' : 'fill'];

  return css`
    display: inline-block;
    height: ${height}px;
    line-height: ${lineHeight}px;
    text-align: center;
    padding: 0 ${padding}px;
    color: ${color};
    font-size: ${fontSize}px;
    background-color: ${backgroundColor};
    border: ${$isStroke ? `1px solid ${borderColor}` : 'none'};
    border-radius: ${height / 2}px;
    white-space: nowrap;
    transform-origin: center center;
    transition: all ${duration.status}ms ${easing.status};
    ${$dot && `width: ${height}px`};
  `;
};

export const StyledBadge = styled.span`
  ${badgePositionCSS}
  ${badgeCoreCSS}
`;

export const StyledIndependentBadge = styled.span`
  ${badgeCoreCSS}
`;

// used for mock font-size less than 12px in chrome
export const StyledBadgeInnerText = styled.span`
  display: inline-block;
  font-style: normal;
  vertical-align: top;
  transform-origin: center center;
`;

export const StyledBadgeContainer = styled.span`
  display: inline-block;
  position: relative;
`;
