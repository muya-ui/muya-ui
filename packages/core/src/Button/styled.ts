import styled, { css } from 'styled-components';

import { StyledCheckboxLabel } from '../Checkbox/styled';
import { IButtonPureProps } from './types';

const baseCSS = css`
  /* 相对定位 */
  position: relative;

  /* flex + inline */
  display: inline-flex;

  /* 内部元素水平&垂直居中 */
  align-items: center;
  justify-content: center;
  vertical-align: middle;
`;

export const StyledButton = styled.button`
  /* 样式重置部分 */
  outline: 0;
  margin: 0;
  font-family: inherit;
  text-transform: none;
  overflow: visible;
  box-sizing: border-box;
  ${baseCSS}
`;

export const StyledLink = styled.a`
  text-decoration: none;
  ${baseCSS}
`;

export const StyledSide = styled.span<IButtonPureProps>`
  display: inline-flex;
  ${props => {
    const { prefixNode, theme, suffixNode } = props;
    const { Button: btnToken } = theme.components;
    if (prefixNode) {
      return css`
        padding-right: ${btnToken.sidePadding}px;
      `;
    } else if (suffixNode) {
      return css`
        padding-left: ${btnToken.sidePadding}px;
      `;
    }
  }}

  ${StyledCheckboxLabel} {
    line-height: unset;
  }
`;
