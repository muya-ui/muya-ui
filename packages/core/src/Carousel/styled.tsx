import styled, { css } from 'styled-components';

import IndexIndicator from './IndexIndicator';
import PagerButton from './PagerButton';
import { ICarouselProps } from './types';

const pagerCss = css`
  display: flex;
  align-items: center;
  justify-content: center;
  position: absolute;
  top: 0;
  height: 100%;
  z-index: 2;
`;

export const StyledIndexIndicator = styled(IndexIndicator)`
  position: absolute;
  left: 0;
  width: 100%;
  display: flex;
  align-items: center;
  box-sizing: border-box;
  ${props => {
    const { indicator } = props.theme.components.Carousel;
    return css`
      bottom: ${indicator.bottom}px;
      padding: ${indicator.padding};
    `;
  }}
`;

export const StyledPagerPrev = styled.div`
  ${pagerCss}
  left: 0;
`;

export const StyledPagerNext = styled.div`
  ${pagerCss}
  right: 0;
`;

export function carouselCss(props: ICarouselProps) {
  const { arrow = 'always', indicator = 'center' } = props;
  let hoverCss;
  if (arrow === 'hover') {
    hoverCss = css`
      & ${PagerButton} {
        opacity: 0;
      }

      &:hover {
        ${PagerButton} {
          opacity: 1;
        }
      }
    `;
  }

  let alignStr = 'center';
  if (indicator === 'left') {
    alignStr = 'flex-start';
  } else if (indicator === 'right') {
    alignStr = 'flex-end';
  }
  return css`
    position: relative;
    ${hoverCss}

    & ${StyledIndexIndicator} {
      justify-content: ${alignStr};
    }
  `;
}
