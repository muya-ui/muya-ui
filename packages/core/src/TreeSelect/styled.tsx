import styled, { css } from 'styled-components';

import { ITheme } from '@muya-ui/theme-light';

import { scrollBarStyle } from '../ScrollView/styled';
import addPx from '../utils/addPx';
import { StyledTree } from '../Tree/styled';

export interface IStyledTreePopupProps {
  theme: ITheme;
  $width: string | number;
  $showEmptyView: boolean;
}

export const StyledTreeScrollContainer = styled.div``;

export const StyledTreePopup = styled.div<IStyledTreePopupProps>`
  ${props => {
    const {
      theme: {
        shadows,
        colors,
        components: { TreeSelect: token },
      },
      $width,
      $showEmptyView,
    } = props;
    let emptyStyle;
    if ($showEmptyView) {
      emptyStyle = css`
        display: flex;
        justify-content: center;
        align-items: center;
        padding-right: 0;
        min-height: ${token.treeHeight}px;
      `;
    }
    return css`
      box-sizing: border-box;
      width: ${addPx($width)};
      box-shadow: ${shadows.pattern.popper.normal};
      padding-right: ${token.popupPadding}px;
      padding-bottom: ${token.popupPadding}px;
      background: ${colors.pattern.background.higher};
      ${emptyStyle};
      ${StyledTreeScrollContainer} {
        max-height: ${token.treeHeight}px;
        overflow: auto;
        ${scrollBarStyle({ theme: props.theme, hideEndButton: true })};
        ${StyledTree} {
          overflow: unset;
        }
      }
    `;
  }}
`;

export const StyledTreeNodeLabelWithIcon = styled.div`
  display: inline-flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
`;
