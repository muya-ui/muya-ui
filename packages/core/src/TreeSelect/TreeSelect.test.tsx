import React from 'react';
import renderer from 'react-test-renderer';

import muyaThemeLight from '@muya-ui/theme-light';

import { TreeSelect } from './index';
import { StyledTreePopup } from './styled';

const treeData = [
  {
    title: '成组条目#0',
    key: '0-0',
    children: [
      {
        title: '成组条目#0#0',
        key: '0-0-0',
        disabled: true,
        children: [
          { title: '普通条目#0#0#0', key: '0-0-0-0' },
          { title: '普通条目#0#0#1', key: '0-0-0-1' },
          { title: '普通条目#0#0#2', key: '0-0-0-2' },
        ],
      },
      {
        title: '成组条目#0#1',
        key: '0-0-1',
        children: [
          { title: '普通条目#0#1#0', key: '0-0-1-0' },
          { title: '普通条目#0#1#1', key: '0-0-1-1' },
          { title: '普通条目#0#1#2', key: '0-0-1-2' },
        ],
      },
      {
        title: '成组条目#0#2',
        key: '0-0-2',
      },
    ],
  },
  {
    title: '成组条目#1',
    key: '0-1',
    children: [
      { title: '普通条目#1#0#0', key: '0-1-0-0' },
      { title: '普通条目#1#0#1', key: '0-1-0-1' },
      { title: '普通条目#1#0#2', key: '0-1-0-2' },
    ],
  },
  {
    title: '成组条目#2',
    key: '0-2',
  },
];

describe('TreeSelect', () => {
  test('测试正常 render', () => {
    const wrapper = renderer
      .create(
        <>
          <TreeSelect id="1" triggerId="1" popupVisible treeData={treeData}></TreeSelect>
          <TreeSelect
            id="2"
            triggerId="2"
            popupVisible
            treeCheckable
            treeData={treeData}
          ></TreeSelect>
          <TreeSelect treeData={[]}></TreeSelect>
        </>,
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  test('测试 empty render', () => {
    const wrapper = renderer
      .create(<StyledTreePopup theme={muyaThemeLight} $showEmptyView={true} $width={240} />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
