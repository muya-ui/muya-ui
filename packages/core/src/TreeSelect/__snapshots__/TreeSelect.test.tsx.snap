// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeSelect 测试 empty render 1`] = `
.c0 {
  box-sizing: border-box;
  width: 240px;
  box-shadow: 0 0 12px 0 rgba(102,102,102,0.12);
  padding-right: 4px;
  padding-bottom: 4px;
  background: rgba(255,255,255,1);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 0;
  min-height: 200px;
}

.c0 .c2 {
  max-height: 200px;
  overflow: auto;
}

.c0 .c2::-webkit-scrollbar {
  width: 4px;
  height: 4px;
}

.c0 .c2::-webkit-scrollbar-thumb {
  border-radius: 999rem;
  background-color: transparent;
}

.c0 .c2:hover::-webkit-scrollbar-thumb {
  background-color: rgba(51,51,51,0.12);
}

.c0 .c2:hover::-webkit-scrollbar-thumb:hover {
  background-color: rgba(51,51,51,0.16);
}

.c0 .c2::-webkit-scrollbar-button:vertical:end:decrement {
  display: block;
  width: 4px;
  height: 4px;
  background-color: transparent;
}

.c0 .c2 .c1 {
  overflow: unset;
}

<div
  className="c0"
/>
`;

exports[`TreeSelect 测试正常 render 1`] = `
Array [
  .c7 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  fill: currentColor;
}

.c6 {
  line-height: 0;
  -webkit-animation: cxSSXZ 150ms linear;
  animation: cxSSXZ 150ms linear;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  cursor: pointer;
}

.c2 {
  width: 240px;
  height: 32px;
  padding: 0 12px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 12px;
  line-height: 16px;
  border-radius: 2px;
  border: 1px solid rgba(221,221,221,1);
  -webkit-transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  vertical-align: middle;
  background-color: rgba(255,255,255,1);
}

.c2:hover {
  border-color: rgba(209,209,209,1);
}

.c4 {
  display: inline-block;
  outline: none;
  border: 0;
  padding: 0;
  margin: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  min-width: 0;
  background: none;
  font-size: 12px;
  line-height: 16px;
  caret-color: rgba(26,122,248,1);
  color: rgba(51,51,51,1);
}

.c4::selection {
  background: rgba(26,122,248,0.16);
}

.c4::-webkit-input-placeholder {
  color: rgba(204,204,204,1);
}

.c4::-moz-placeholder {
  color: rgba(204,204,204,1);
}

.c4:-ms-input-placeholder {
  color: rgba(204,204,204,1);
}

.c4::placeholder {
  color: rgba(204,204,204,1);
}

.c5 {
  margin-left: 12px;
  line-height: 0;
  color: rgba(51,51,51,1);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  outline: none;
  cursor: default;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: rgba(204,204,204,1);
  cursor: pointer;
}

.c5:hover {
  color: rgba(187,187,187,1);
}

.c5:active {
  color: rgba(153,153,153,1);
}

.c0 {
  position: relative;
  display: inline-block;
  outline: none;
}

.c0 .c3 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c0 .c10,
.c0 .c1 {
  border-color: rgba(26,122,248,1);
}

.c0 .c1,
.c0 .c10 {
  cursor: pointer;
}

.c0 .c1 input,
.c0 .c10 input,
.c0 .c1 svg,
.c0 .c10 svg {
  cursor: pointer;
}

.c8 .c3 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c8 .c10,
.c8 .c1 {
  border-color: rgba(26,122,248,1);
}

.c9 .c3 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c9 .c1,
.c9 .c10 {
  cursor: pointer;
}

.c9 .c1 input,
.c9 .c10 input,
.c9 .c1 svg,
.c9 .c10 svg {
  cursor: pointer;
}

<div
    className="c0 muya-tree-select-input-wrapper"
    id="1"
    role="combobox"
    tabIndex={0}
  >
    <div
      className="c1 c2 muya-input-input-wrapper"
      disabled={false}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      size="m"
    >
      <input
        aria-describedby="1"
        autoFocus={false}
        className="c3 c4 muya-input-input"
        disabled={false}
        id="1"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        onKeyDown={[Function]}
        placeholder="请选择"
        readOnly={true}
        size="m"
        type="text"
        value=""
      />
      <div
        className="c5 muya-input-node-wrapper"
        size="m"
      >
        <div
          className="c6"
        >
          <svg
            aria-hidden="false"
            className="c7"
            focusable="false"
            style={
              Object {
                "color": "rgba(204,204,204,1)",
                "height": 8,
                "width": 8,
              }
            }
            viewBox="0 0 1024 1024"
          >
            <path
              d="M832.267 277.333H189.46a21.333 21.333 0 00-17.685 33.259l306.09 453.76a21.333 21.333 0 0034.817.779L849.387 311.36a21.333 21.333 0 00-17.12-34.027z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  .c14 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  fill: currentColor;
}

.c13 {
  line-height: 0;
  -webkit-animation: cxSSXZ 150ms linear;
  animation: cxSSXZ 150ms linear;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  cursor: pointer;
}

.c4 {
  position: relative;
  box-sizing: border-box;
  overflow-x: hidden;
  overflow-y: auto;
  outline: none;
}

.c6 {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 {
  display: inline-block;
  overflow: hidden;
  background: transparent;
  border-width: 0;
  margin: 0;
  padding: 0;
  outline: 0;
  margin-bottom: 4px;
}

.c10 {
  position: absolute;
  top: 0;
  left: 0;
  visibility: hidden;
  height: 0;
  overflow: scroll;
  white-space: pre;
  line-height: 1;
}

.c12 {
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  margin: auto;
}

.c2 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  overflow: hidden;
  outline: none;
  box-sizing: border-box;
  width: 240px;
  border: 1px solid rgba(221,221,221,1);
  box-shadow: none;
  border-radius: 2px;
  background: rgba(255,255,255,1);
  line-height: 1;
  font-size: 12px;
  -webkit-transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  cursor: text;
}

.c2:hover {
  border-color: rgba(209,209,209,1);
}

.c2 .c3 {
  padding-left: 12px;
  padding-right: 24px;
  max-height: 63px;
}

.c2 .c3::-webkit-scrollbar {
  width: 4px;
  height: 4px;
}

.c2 .c3::-webkit-scrollbar-thumb {
  border-radius: 999rem;
  background-color: transparent;
}

.c2 .c3:hover::-webkit-scrollbar-thumb {
  background-color: rgba(51,51,51,0.12);
}

.c2 .c3:hover::-webkit-scrollbar-thumb:hover {
  background-color: rgba(51,51,51,0.16);
}

.c2 .c3::-webkit-scrollbar-button:vertical:end:decrement {
  display: block;
  width: 4px;
  height: 4px;
  background-color: transparent;
}

.c2 .c11 {
  right: 12px;
}

.c2 .c7 {
  width: 1px;
  color: rgba(102,102,102,1);
}

.c2 input[readonly] {
  cursor: pointer;
}

.c2 input::-webkit-input-placeholder {
  color: rgba(204,204,204,1);
}

.c2 input::-moz-placeholder {
  color: rgba(204,204,204,1);
}

.c2 input:-ms-input-placeholder {
  color: rgba(204,204,204,1);
}

.c2 input::placeholder {
  color: rgba(204,204,204,1);
}

.c2 .c9 {
  font-size: 12px;
}

.c2 .c5 {
  height: 30px;
}

.c2 .c5 .c7 {
  margin-bottom: 0;
}

.c2 .c17 {
  max-width: 208px;
}

.c15 .c1,
.c15 .c18 {
  border-color: rgba(26,122,248,1);
}

.c15 .c18,
.c15 .c1 {
  cursor: pointer;
}

.c15 .c18 input,
.c15 .c1 input,
.c15 .c18 svg,
.c15 .c1 svg {
  cursor: pointer;
}

.c0 {
  position: relative;
  display: inline-block;
  outline: none;
}

.c0 .c19 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c0 .c1,
.c0 .c18 {
  border-color: rgba(26,122,248,1);
}

.c16 .c18,
.c16 .c1 {
  cursor: pointer;
}

.c16 .c18 input,
.c16 .c1 input,
.c16 .c18 svg,
.c16 .c1 svg {
  cursor: pointer;
}

<div
    className="c0 muya-tree-select-input-wrapper"
    id="2"
    role="combobox"
    tabIndex={0}
  >
    <div
      className="c1 c2 muya-tags-input-input-wrapper"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="c3 c4 muya-tags-input-content-wrapper"
      >
        <div
          className="c5 c6 muya-tags-input-tags-wrapper"
        >
          <input
            aria-describedby="2"
            autoFocus={false}
            className="c7 c8 muya-tags-input-input"
            disabled={false}
            id="2"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onInput={[Function]}
            onKeyDown={[Function]}
            placeholder="请选择"
            readOnly={false}
            type="text"
            value=""
          />
          <div
            className="c9 c10"
          >
            请选择
          </div>
        </div>
      </div>
      <div
        className="c11 c12"
      >
        <div
          className="c13"
        >
          <svg
            aria-hidden="false"
            className="c14"
            focusable="false"
            style={
              Object {
                "color": "rgba(204,204,204,1)",
                "height": 8,
                "width": 8,
              }
            }
            viewBox="0 0 1024 1024"
          >
            <path
              d="M832.267 277.333H189.46a21.333 21.333 0 00-17.685 33.259l306.09 453.76a21.333 21.333 0 0034.817.779L849.387 311.36a21.333 21.333 0 00-17.12-34.027z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  .c7 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  fill: currentColor;
}

.c6 {
  line-height: 0;
  -webkit-animation: bFXLB 150ms linear;
  animation: bFXLB 150ms linear;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  cursor: pointer;
}

.c2 {
  width: 240px;
  height: 32px;
  padding: 0 12px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 12px;
  line-height: 16px;
  border-radius: 2px;
  border: 1px solid rgba(221,221,221,1);
  -webkit-transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  vertical-align: middle;
  background-color: rgba(255,255,255,1);
}

.c2:hover {
  border-color: rgba(209,209,209,1);
}

.c4 {
  display: inline-block;
  outline: none;
  border: 0;
  padding: 0;
  margin: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  min-width: 0;
  background: none;
  font-size: 12px;
  line-height: 16px;
  caret-color: rgba(26,122,248,1);
  color: rgba(51,51,51,1);
}

.c4::selection {
  background: rgba(26,122,248,0.16);
}

.c4::-webkit-input-placeholder {
  color: rgba(204,204,204,1);
}

.c4::-moz-placeholder {
  color: rgba(204,204,204,1);
}

.c4:-ms-input-placeholder {
  color: rgba(204,204,204,1);
}

.c4::placeholder {
  color: rgba(204,204,204,1);
}

.c5 {
  margin-left: 12px;
  line-height: 0;
  color: rgba(51,51,51,1);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  outline: none;
  cursor: default;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: rgba(204,204,204,1);
  cursor: pointer;
}

.c5:hover {
  color: rgba(187,187,187,1);
}

.c5:active {
  color: rgba(153,153,153,1);
}

.c8 .c3 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c8 .c10,
.c8 .c1 {
  border-color: rgba(26,122,248,1);
}

.c8 .c1,
.c8 .c10 {
  cursor: pointer;
}

.c8 .c1 input,
.c8 .c10 input,
.c8 .c1 svg,
.c8 .c10 svg {
  cursor: pointer;
}

.c9 .c3 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c9 .c10,
.c9 .c1 {
  border-color: rgba(26,122,248,1);
}

.c0 {
  position: relative;
  display: inline-block;
  outline: none;
}

.c0 .c3 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c0 .c1,
.c0 .c10 {
  cursor: pointer;
}

.c0 .c1 input,
.c0 .c10 input,
.c0 .c1 svg,
.c0 .c10 svg {
  cursor: pointer;
}

<div
    className="c0 muya-tree-select-input-wrapper"
    role="combobox"
    tabIndex={0}
  >
    <div
      className="c1 c2 muya-input-input-wrapper"
      disabled={false}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      size="m"
    >
      <input
        aria-describedby={null}
        autoFocus={false}
        className="c3 c4 muya-input-input"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        onKeyDown={[Function]}
        placeholder="请选择"
        readOnly={true}
        size="m"
        type="text"
        value=""
      />
      <div
        className="c5 muya-input-node-wrapper"
        size="m"
      >
        <div
          className="c6"
        >
          <svg
            aria-hidden="false"
            className="c7"
            focusable="false"
            style={
              Object {
                "color": "rgba(204,204,204,1)",
                "height": 8,
                "width": 8,
              }
            }
            viewBox="0 0 1024 1024"
          >
            <path
              d="M832.267 277.333H189.46a21.333 21.333 0 00-17.685 33.259l306.09 453.76a21.333 21.333 0 0034.817.779L849.387 311.36a21.333 21.333 0 00-17.12-34.027z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
]
`;
